nengo/networks/assoc_mem.py

```diff
@@ -75,9 +75,12 @@ def __init__(self, input_vectors, output_vectors=None,  # noqa: C901
         if is_iterable(output_vectors):
             output_vectors = np.matrix(output_vectors)
 
-        # Fail if number of input items and number of output items don't
-        # match
-        if input_vectors.shape[0] != output_vectors.shape[0]:
+        if input_vectors.shape[0] <= 0:
+            # Fail if number of input items <= 0
+            raise ValueError('Number of input vectors cannot be <= 0.')
```

Comment:
It might also help to add an `is_array` check. And, if you really want it to be a `np.matrix`, you should make it a matrix if you get an array; right now, you might get different behavior if you pass in a list versus an equivalent np.array of that list.